python/tests/test_topology.py

```diff
@@ -220,24 +217,24 @@ def update_kc_vectors_single_leaf(kc_vecs, u, time, sample_index_map):
 
 
 def update_kc_vectors_all_pairs(tree, kc_vecs, c1, c2, depth, time, sample_index_map):
-    leaves = tree.tree_sequence.samples()
+    samples = tree.tree_sequence.samples()
     c1_left = tree.left_sample(c1)
     c1_right = tree.right_sample(c1)
-    leaf1_index = c1_left
+    sample1_index = c1_left
     while True:
         c2_left = tree.left_sample(c2)
         c2_right = tree.right_sample(c2)
-        leaf2_index = c2_left
+        sample2_index = c2_left
         while True:
-            leaf1 = leaves[leaf1_index]
-            leaf2 = leaves[leaf2_index]
+            leaf1 = samples[sample1_index]
+            leaf2 = samples[sample2_index]
             update_kc_vectors_pair(kc_vecs, leaf1, leaf2, depth, time, sample_index_map)
```

Comment:
Can't you pass ``sample1_index`` and ``sample2_index`` in directly here, and avoid using ``sample_index_map`` in update_kc_vectors_pair? Looks like we're going index -> node ID -> index here.

Reply:
Definitely, fixed this in C and forgot to do the same in python.